tools/release.sh

```diff
@@ -0,0 +1,237 @@
+        echo "'bintrayUser' is not set in the '~/.gradle/gradle.properties'."
+        exit -1
+    fi
+
+    if ! grep "bintrayKey=.*" $HOME/.gradle/gradle.properties > /dev/null; then
+        echo "'bintrayKey' is not set in the '~/.gradle/gradle.properties'."
+        exit -1
+    fi
+
+    # Check gradle params
+    if grep buildTargetABIs $HOME/.gradle/gradle.properties | grep -v "^#" > /dev/null ; then
+        echo "'buildTargetABIs' should be disabled in the '$HOME/.gradle/gradle.properties'."
+        exit -1
+    fi
+    if grep ccachePath $HOME/.gradle/gradle.properties | grep -v "^#" > /dev/null ; then
+        echo "'ccachePath' should be disabled in the '$HOME/.gradle/gradle.properties'."
+        exit -1
+    fi
+    if grep lcachePath $HOME/.gradle/gradle.properties | grep -v "^#" > /dev/null ; then
+        echo "'lcachePath' should be disabled in the '$HOME/.gradle/gradle.properties'."
+        exit -1
+    fi
+
+    if [[ -z ${REALM_IO_PATH} ]] ; then
+        REALM_IO_PATH="$(pwd)/../realm.io"
+    fi
+    if [[ ! -e ${REALM_IO_PATH} ]] ; then
+        echo "Please set 'REALM_IO_PATH' to the 'realm.io' repository path to publish javadoc."
+        exit -1
+    fi
+}
+
+prepare_branch() {
+    echo "Preparing release branch..."
+
+    git fetch --all
+    git checkout releases
+    git reset --hard origin/releases
+    if [[ "$BRANCH_TO_RELEASE" != "releases" ]] ; then
+        echo "Releasing from other branches than 'releases' is not supported right now."
+        exit -1
+    fi
+
+    git clean -xfd
+    git submodule update --init --recursive
+
+    if ! cat version.txt | grep -q "SNAPSHOT" ; then
+        echo "'version.txt' doesn't contain 'SNAPSHOT'."
+        exit -1
+    fi
+
+    version_in_changelog=$(head -1 CHANGELOG.md | grep -o "[0-9]*\.[0-9]*\.[0-9]*")
+    VERSION=$(cat version.txt | grep -o "[0-9]*\.[0-9]*\.[0-9]*")
+    if [[ "${VERSION}" != "${version_in_changelog}" ]] ; then
+        echo "'version.txt' doens't match the entry in 'CHANGELOG.md'. ${VERSION} vs ${version_in_changelog}."
+        exit -1
+    fi
+
+    # Check if tag exists in remote
+    if git ls-remote --tags origin | grep v${VERSION} > /dev/null ; then
+        echo "Tag 'v${VERSION}' exists in remote!"
+        exit -1
+    fi
+    if git tag | grep v${VERSION} > /dev/null ; then
+        git tag -d v${VERSION}
+    fi
+
+    # Update date in change log
+    cur_date=$(date "+%F")
+    sed -i "1 s/YYYY-MM-DD/${cur_date}/" CHANGELOG.md
+    git add CHANGELOG.md
+    git commit -m "Update changelog date"
+
+    # This will create 2 new commits to change the version.txt. The top one is the next release version + SNAPSHOT.
+    ./gradlew release
+    # Checkout the one with current version number.
+    git checkout HEAD~1
+}
+
+build() {
+    echo "Building..."
+
+    ./gradlew assemble
+
+    echo "Verifying examples..."
+
+    check_adb_device
+
+    # Verify examples
+    cd examples
```

Comment:
Maybe `(cd examples && ./gradlew uninstallAll && ./gradlew monkeyDebug)` and no `cd ..`